src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +31,19 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        splitAddressIntoParts(address);
+    }
+    
+    /**
+     * Splits the Address into its four different parts
+     */
+    
+    private void splitAddressIntoParts(String address){
```

Comment:
- Did you make sure an array index out of bound is not thrown when running the test-cases.

- Can handle this without forcing 4 address components for the user. Think about it too.